cartographer/mapping/internal/3d/pose_graph_3d.cc

```diff
@@ -560,7 +564,7 @@ void PoseGraph3D::WaitForAllComputations() {
                                   absl::FromChrono(common::FromSeconds(1.)))) {
     report_progress();
   }
-  CHECK_EQ(constraint_builder_.GetNumFinishedNodes(), num_trajectory_nodes);
+  // CHECK_EQ(constraint_builder_.GetNumFinishedNodes(), num_trajectory_nodes);
```

Comment:
Maybe this `CHECK` can be moved into `RunFinalOptimization`.

Reply:
I see this as a warning as it could mean that implications of the change are not well understood and that it was developed in this fashion:
1. change the code to clear the queue
2. test
3. got the assertion
4. comment the assertion
5. test
6. ok

This is just to let you know how it can be interpreted as I think that you are aware of implications of your code change.

Reply:
You are right it is the case that I stumbled into the assertion after I implemented clearing the work queue, and that this was "developed" in such a way. This does not preclude understanding (or trying to) what is going on and trying to come up with the ideal solution :-) This was more intended to be an RFC (hence my "maybe we could move this..." comment) rather than the definite proposal.

I am also not quite happy with this because even with this change, we can still end up waiting for some time because the constraint builder has its own set of scheduled tasks that can be blocking destruction.

Reply:
@ojura Code in comments is bad and should rather be removed completely. That said, if you would like this merged it would need to be a PR intended for review, not an RFC as PR. That is you should have thought through the implications and be able to explain why this is fine the way you implemented it. Please let me know if you want to do this. Otherwise, I would close the PR.

Reply:
@wohe in that case the PR should be closed because it's not ready for review prior to merge; and an issue or RFC opened for this, perhaps referencing the changes here as a starting point.

Reply:
@wohe closed and opened #1721 referencing this.